File: src/main/java/cherkasov/com/TaskEntity.java
package cherkasov.com;

/**
 * The entity that stores address for downloading files and file name
 * to store on filesystem
 * It is used to represent tasks in the task queue
 */

public class TaskEntity {
    private final String url;
    private final String fileName;

    public TaskEntity(String url, String fileName) {
        this.url = url;
        this.fileName = fileName;
    }

    public String getUrl() {
        return url;
    }

    public String getFileName() {
        return fileName;
    }

    /**
     * Uses for unit testing
     */
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TaskEntity that = (TaskEntity) o;

        if (!getUrl().equals(that.getUrl())) return false;
        return getFileName().equals(that.getFileName());
    }
}

File: src/main/java/cherkasov/com/TaskEntity.kt
package cherkasov.com

/**
 * The entity that stores address for downloading files and file name
 * to store on filesystem
 * It is used to represent tasks in the task queue
 */
class TaskEntity(val url: String, val fileName: String) {

    /**
     * Uses for unit testing
     */
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as TaskEntity
        return if (url != that.url) false else fileName == that.fileName
    }

}